rt/features/logging/src/main/java/org/apache/cxf/ext/logging/LoggingOutInterceptor.java

```diff
@@ -82,7 +82,10 @@ private OutputStream createCachingOut(Message message, final OutputStream os, Ca
             newOut.setThreshold(threshold);
         }
         if (limit > 0) {
-            newOut.setCacheLimit(limit);
+            // make the limit for the cache greater than the limit for the truncated payload in the log event, 
+            // this is necessary for finding out that the payload was truncated 
+            //(see boolean isTruncated = cos.size() > limit && limit != -1;)  in method copyPayload
+            newOut.setCacheLimit(limit + 1);
```

Comment:
instead of just setting the limit to 'limit +1' one should check whether the limit == Integer.MAX_VALUE and if this is the case just use Integer.MAX_VALUE